ocaml/xapi/message_forwarding.ml

```diff
@@ -4297,6 +4297,17 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
   end
 
   module Cluster = struct
+
+    let forward_cluster_op ~local_fn ~__context op =
+      let localhost = Helpers.get_localhost ~__context in
+      let is_local_cluster_host self = Db.Cluster_host.get_host ~__context ~self = localhost in
+      let host = match Db.Cluster_host.get_all ~__context with
+        | [] -> Helpers.get_master ~__context
+        | lst when List.exists is_local_cluster_host lst -> localhost
```

Comment:
Just a nitpick: this matches a list of hosts but this is called `lst` here and `self` below which is a bit confusing. I would have called this match `hosts` and below would have used `host::_`.

Reply:
I've changed it to use `cluster_host` and `cluster_hosts` (to avoid confusion with the `host` object which is different from a `Cluster_host` object)